### DISCUSSION TITLE
How to declare a method that uses a colon?

### DISCUSSION THREAD

**Discussion by maikawk** [SquidDev/urn#36] [Q&A] [answered]
I am using a Lua 5.1 compiler which has several native objects that have methods that use colons. for example
```
local value = nativeObject:GenerateValue("someParameter")
```
So I would like to do this in urn:
```
(define value (GenerateValue "someParameter"))
```
But I don't know exactly how to do it. try to create a nativeObject.lisp file with this:
```
(define-native someParameter)
```
but I don't know what your meta file should look like.

**Comment by SquidDev** [comment, marked as answer]
The [`self`](https://urn-lang.com/docs/lib.data.function.html#self-x-key-args) method can be used to call functions in a self-call (i.e. with a colon):

So your first example would become:

```lisp
(with (value (self nativeObject :GenerateValue "someParameter"))
  ;; ...
```

I'm afraid I don't follow the other two code blocks - not sure what you're trying to demonstrate there.
